modules/basic-skills/src/views/full/callApi.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import { Row, Col, Label, Input } from 'reactstrap'
+import Select from 'react-select'
+import style from './style.scss'
+import { BotpressTooltip } from 'botpress/tooltip'
+import { LinkDocumentationProvider } from 'botpress/documentation'
+
+const methodOptions = [
+  { label: 'get', value: 'get' },
+  { label: 'post', value: 'post' },
+  { label: 'put', value: 'put' },
+  { label: 'delete', value: 'delete' }
+]
+
+const memoryOptions = [
+  { label: 'Temp', value: 'temp' },
+  { label: 'Session', value: 'session' },
+  { label: 'Bot', value: 'bot' },
+  { label: 'User', value: 'user' }
+]
+
+export class CallAPI extends React.Component {
+  state = {
+    selectedMethod: methodOptions[0],
+    selectedMemory: memoryOptions[0],
+    variable: 'response',
+    body: undefined,
+    url: undefined
+  }
+
+  componentDidMount() {
+    const data = this.props.initialData
+    if (data) {
+      this.setState({
+        selectedMethod: { value: data.method, label: data.method },
+        selectedMemory: { value: data.memory, label: data.memory },
+        variable: data.variable,
+        body: data.body,
+        url: data.url
+      })
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.url && this.state.selectedMethod) {
```

Comment:
I suggest inverting the condition so one nesting level can be removed for the whole method. Also, what do you think of moving the logic in its own method? 

I prefer when we just keep the logic of "did it updated" here, and move the actions in its own method, like 
```js
componentDidUpdate(){
  if(this.state.url && this.state.selectedMethod){
    this.updateApiParams()
  }
}
```

What is even better, is to call the actions only when there is indeed a change
```js
componentDidUpdate(prevProps, prevState){
  if(this.state.url && this.state.selectedMethod && this.state !== prevState){
    this.updateApiParams()
  }
}
```